### PR TITLE
Disambiguate repositories by their owner

### LINKED ISSUE #1848: Disambiguate repositories with the same name
Repositories are named by their GitHub repository or the on-disk directory if they can't be associated with a GitHub repository. This can pretty easily lead to naming collisions.

Some prior art:

ad77f1b0-7cb7-11e4-846e-c62f6005509b-1

In this case, we have multiple clones of the same repository. So we start by disambiguating by GitHub owner name (`robrix`) and then, since that's still ambiguous, we append the containing directory name (`External` vs. `Swift`).

### PR DESCRIPTION
**Pull request by joshaber**
Fixes #1848

screen shot 2017-06-16 at 2 23 27 pm

### CODE REVIEW

File: app/src/ui/repositories-list/group-repositories.ts [hunk 1]
```diff
@@ -44,12 +45,22 @@ export function groupRepositories(repositories: ReadonlyArray<Repositoryish>): R
...
+      names.set(repository.name, existingCount + 1)
+    }
+
     repositories.sort((x, y) => caseInsensitiveCompare(x.name, y.name))
-    const items = repositories.map(r => ({
-      text: r.name,
-      id: r.id.toString(),
-      repository: r,
-    }))
+    const items: ReadonlyArray<IRepositoryListItem> = repositories.map(r => {
```
Comment (shiftkey) [inline app/src/ui/repositories-list/group-repositories.ts]: I _think_ we can drop the type here

Comment (joshaber) [inline app/src/ui/repositories-list/group-repositories.ts]: We _can_, but having the type annotation makes the compile errors much clearer when we change `IRepositoryListItem`.

**Comment by joshaber** [comment]
🏥

File: app/src/ui/repositories-list/group-repositories.ts (hunk 1 above)
Comment (shiftkey) [inline app/src/ui/repositories-list/group-repositories.ts]: :thumbsup:

**Pull request merged**